commcare_connect/users/views.py

```diff
@@ -44,3 +49,22 @@ def get_redirect_url(self):
 
 
 user_redirect_view = UserRedirectView.as_view()
+
+
+class CreateUserLinkView(ClientProtectedResourceMixin, APIView):
+    parser_classes = [parsers.JSONParser]
+
+    def post(self, request):
+        commcare_username = request.data.get('commcare_username')
+        connect_username = request.data.get('connect_username')
+        if not commcare_username or not connect_username:
+            return Response("commcare_username and connect_username required", status=status.HTTP_400_BAD_REQUEST)
+        try:
+            user = User.objects.get(username=connect_username)
```

Comment:
This relies on the user having authenticated at least once with connect before this call, which I think is a fair assumption, but one I am willing to revisit.

Reply:
`User` model uses the `email` field to identify users so I think this `username` field might not be populated because the `username` field type is set to None in the User model.

Reply:
Sorry, I didn't realize that, thanks for flagging. I can wait for @snopoke to weigh in on how he imagined this working since he seems to have done the initial implementation there, but that seems unfortunate to me since the mobile users won't have an email, and instead have a ConnectID username/userid as their primary identifier, which I was envisioning using in the username here (following django-oauth-toolkit standard for token introspection).

@snopoke were you picturing an entirely separate user model for mobile users of connect?

Reply:
This project was built using the Django cookiecutter which defaults to using email for auth (which is the pattern for most web apps). We can easily switch to supporting both:

1. Add back the `username` field to the user model
2. Change the `ACCOUNT_AUTHENTICATION_METHOD` setting to `username_email`

We may have to tweak a [few other settings](https://github.com/dimagi/commcare-connect/blob/ac92c81313c50311065e430cc3441187c1872b30/config/settings/base.py#L245-L255) as well.

Reply:
I started making changes on the token introspection PR since the user creation in that workflow requires username. I got most of the tests passing, but have one to go. Will pick it up again tomorrow, and do more proper localtesting, which I have yet to do.